packages/website/docs/TextInput.md

```diff
@@ -1,20 +1,46 @@
-## Guidelines
+## Best Practices
 
-### Restrictions & Size
+Use text inputs when the user must enter one **short line of text.**
 
--   Users should be informed of any input restrictions, such as restrictions on character types or string length.
--   The text input area width should suggest the expected size of the user input. Try to forsee the text input length so as to avoid cutting it off. The text input area should be large enough to display the whole text input in most cases.
--   Avoid setting a character limit, or make sure to set it high enough to avoid constraining most user entries. If this is not possible, let users know how many characters are left.
--   Ensure that users with small screens can still enter all required information.
+The width of the text input should be proportional to the expected user input and large enough to display the entire text.
 
-### Behaviour
+## Labeling
 
--   Display an inline validation message when the user is done interacting with the input.
+Keep labels short, **under three words**.
+
+Use descriptive labels without action verbs.
+
+-   E.g. Use "Email" instead of "Write your email".
+
+## Help text and instructions
+
+Help text provide extra guidance on **what** information must be provided or **how** the user should fill it in (such as restricted characters).
+
+Help text should be **short one-liner**.
+There are two ways to provide help text guidance:
+
+-   Written instructions : "Provide your business email."
+-   Example given : "E.g.: johndoe@acme.com"
+
+A combination of both is also acceptable.
+
+The label of the text input should be self-explanatory as much as possible.
+Add a title and description when providing additional guidance is critical to the user to understand why the information is required or how the information will be used.
+
+## Feedback and validation
+
+If some characters are restricted, consider indicating it in an help text instead of relying only on the validation message to inform the user about them.
+
+If the provided information isn't compliant with the requirements, make sure that the validation feedback clearly indicates how to fix it in the text of the validation message.
+
+If the information is optional, the text input will be automatically tagged as so to indicate that it can be left blank.
```

Comment:
@kpoulincoveo It's unclear to me what we mean here.
"If the information is optional" = if it's optional that the user provides text, right?
"the text input will automatically be tagged" = the placeholder contains `(optional)`? 🤔